src/rules/no-unsafe-values.js

```diff
@@ -3,6 +3,10 @@
  * @author Bradley Meck Farias
  */
 
+// These numbers evaluated to zero unexpectedly.  If the group's digits
+// are all zero, that's fine.
+const SMALL_NUMBERS = [/\.(\d+)/u, /((?:\d+\.)?\d+)e/iu];
```

Comment:
Do we need to match the entire number sequence? If not, can you add some comments explaining what patterns these regexes are meant to match?
```suggestion
const SMALL_NUMBERS = [/^\.(\d+)$/u, /^((?:\d+\.)?\d+)e$/iu];
```

Reply:
Those are intentionally not anchored.  I'll add some comments and examples inline there to make it more clear.

Reply:
I've just rewritten this whole section to make it more clear and less error-prone, I hope.  I just translated the ABNF for `number` into an anchored regex.